web/src/lib/components/user-settings-page/appearance-settings.svelte

```diff
@@ -1,11 +1,71 @@
 <script lang="ts">
   import { fade } from 'svelte/transition';
-  import { colorTheme } from '../../stores/preferences.store';
-  import SettingSwitch from '../admin-page/settings/setting-switch.svelte';
+  import { colorTheme, locale } from '../../stores/preferences.store';
+  import Locales from '$lib/assets/locales.json';
+  import type { ComboBoxOption } from '$lib/components/shared-components/combobox.svelte';
+  import { onMount } from 'svelte';
+  import { findLocale } from '$lib/utils';
+  import { fallbackLocale } from '$lib/constants';
+  import SettingCombobox from '$lib/components/shared-components/settings/setting-combobox.svelte';
+  import SettingSwitch from '$lib/components/shared-components/settings/setting-switch.svelte';
 
-  export const handleToggle = () => {
+  let time = new Date();
+
+  $: formattedDate = time.toLocaleString(editedLocale, {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+  });
+  $: timePortion = time.toLocaleString(editedLocale, {
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+  });
+  $: selectedDate = `${formattedDate} ${timePortion}`;
+  $: editedLocale = findLocale($locale).code;
+  $: selectedOption = {
+    value: findLocale(editedLocale).code || fallbackLocale.code,
+    label: findLocale(editedLocale).name || fallbackLocale.name,
+  };
+
+  onMount(() => {
+    const interval = setInterval(() => {
+      time = new Date();
+    }, 1000);
+
+    return () => {
+      clearInterval(interval);
+    };
+  });
+
+  const getAllLanguages = (): ComboBoxOption[] => {
+    const testNumber = 10;
+    // test locales supported by the browser
+    return Locales.map((locale) => ({
+      label: locale.name,
+      value: locale.code,
+    })).filter((locale) => {
+      try {
+        if (testNumber.toLocaleString(locale.value)) {
+          return locale;
+        }
+      } catch {
+        // do nothing
+      }
+    });
```

Comment:
From my testing in Firefox and Chrome, the browser default is used as a fallback when an invalid locale is provided. That means the filter does effectively nothing.

You could revert to the old behavior of showing all locales, or use a built-in function to determine if a locale is supported like this:

```suggestion
    return Locales.filter(({ code }) => Intl.NumberFormat.supportedLocalesOf(code).length > 0).map((locale) => ({
      label: locale.name,
      value: locale.code,
    }));
```

Reply:
Thanks, I really like it !